buckets/archive/tracker.go

```diff
@@ -182,27 +189,40 @@ func (t *Tracker) trackArchiveProgress(
 		// if error specifies that the auth token isn't found, powergate must have been reset.
 		// return the error as fatal so the archive will be untracked
 		if strings.Contains(err.Error(), "auth token not found") {
-			return false, "", err
+			return 0, "", err
 		}
-		return true, fmt.Sprintf("getting current job %s for bucket %s: %s", jid, buckKey, err), nil
-	}
-
-	if !isJobStatusFinal(res.StorageJob.Status) {
-		return true, "no final status yet", nil
+		return t.jobPollIntervalSlow, fmt.Sprintf("getting current job %s for bucket %s: %s", jid, buckKey, err), nil
 	}
 
-	// Step 2: On success, save Deal data in the underlying Bucket thread. On
-	// failure save the error message. Also update status on Mongo for the archive.
+	// Step 2: On job success, save Deal data in the underlying Bucket thread. On
+	// failure save the error message.
 	if res.StorageJob.Status == userPb.JobStatus_JOB_STATUS_SUCCESS {
-		if err := t.saveDealsInArchive(ctx, buckKey, dbID, dbToken, powInfo.Token, bucketRoot); err != nil {
-			return true, fmt.Sprintf("saving deal data in archive: %s", err), nil
+		if err := t.saveDealsInBucket(ctx, buckKey, dbID, dbToken, powInfo.Token, bucketRoot); err != nil {
+			return t.jobPollIntervalSlow, fmt.Sprintf("saving deal data in archive: %s", err), nil
 		}
 	}
+	// Step 3: Update status on Mongo for the archive.
 	if err := t.updateArchiveStatus(ctx, buckKey, res.StorageJob, false, ""); err != nil {
-		return true, fmt.Sprintf("updating archive status: %s", err), nil
+		return t.jobPollIntervalSlow, fmt.Sprintf("updating archive status: %s", err), nil
+	}
+
+	rescheduleDuration := t.jobPollIntervalFast
+	message := "non-final status"
+	if isJobStatusFinal(res.StorageJob.Status) {
+		message = "reached final status"
+		rescheduleDuration = 0
+	}
+
+	if rescheduleDuration > 0 {
+		for _, dealInfo := range res.StorageJob.DealInfo {
+			if dealInfo.StateId == storagemarket.StorageDealSealing {
```

Comment:
If we are going to reschedule, inspect the deal states and decide if we should use short or long duration.

I really hate the use of `storagemarkets` here, but it's necessary since we aren't modeling those states ourself at any point in the stack. Maybe we should so we can avoid this and have typed states for deals.

Reply:
Agree. We should eventually have our own states in the API.
Even if we use `storagemarket` here, there's the chance that `storagemarket` version used in Hub might diverge from the one returned by Powergate, if both are running different versions that doesn't break the API. (In fact, that was already the case for a long time since we didn't break APIs in Powergate and didn't update the Powergate client in Hub).

To be fair, I see a really tiny chance that the market's code will be changing the numbers of statuses, so isn't a panicking situation. But I completely agree that the API should be self-contained... not only because is nicer, but because is safer.

Might worth creating an issue?

Reply:
Yes, done: https://github.com/textileio/powergate/issues/715